### PR TITLE
Update bdk, maia and bitcoin

### PR DESCRIPTION
**Pull request by klochowicz**
I based this PR on the direction: https://github.com/comit-network/maia/pull/38/files (after first rolling out a backwards-compatible version of `maia`).

One thing that I'm not sure whether it was good, was the change from `schnorr` to `ecdsa` - does this make sense at all? or did I take a wrong turn somewhere?

### CODE REVIEW

**Review by bonomat — APPROVED** [review]
I'd say if it compiles and all tests run the upgrade should be fine.

File: daemon-tests/src/mocks/wallet.rs [hunk 1]
```diff
@@ -82,7 +81,8 @@ pub trait Wallet {
 /// tells the user they have plenty of moneys
 fn dummy_wallet_info() -> Result<WalletInfo> {
     let s = Secp256k1::new();
-    let public_key = ecdsa::PublicKey::new(s.generate_keypair(&mut thread_rng()).1);
+    let public_key =
+        bdk::bitcoin::util::key::PublicKey::new(s.generate_keypair(&mut thread_rng()).1);
```
Comment (luckysori) [inline daemon-tests/src/mocks/wallet.rs]: 🙃 Same thing.

```suggestion
        bdk::bitcoin::PublicKey::new(s.generate_keypair(&mut thread_rng()).1);
```

**Review by luckysori — APPROVED** [review]
Looks good! Good job saving us from breaking changes :D

The change from `schnorrsig::PublicKey` to `XOnlyPublicKey` is fine because it's the same type that has been moved around.

**Pull request merged**